kratos/expression/variable_expression_data_io.cpp

```diff
@@ -230,6 +483,24 @@ void VariableExpressionDataIO<Matrix>::Read(
     }
 }
 
+template<>
+void VariableExpressionDataIO<std::vector<Matrix>>::Read(
+    RawLiteralFlatExpression& rExpression,
+    const IndexType EntityIndex,
+    const std::vector<Matrix>& Value) const
+{
+    const auto& r_shape = rExpression.GetItemShape();
+    const IndexType matrix_size = r_shape[1] * r_shape[2];
+    const IndexType entity_data_begin_index = EntityIndex * r_shape[0] * matrix_size;
+
+    IndexType local_index = 0;
+    for (IndexType i = 0; i < r_shape[0]; ++i) {
+        for (IndexType j = 0; j < matrix_size; ++j) {
```

Comment:
`i` and `j` are super inappropriate index names here. `i` refers to the index of the matrix (as opposed to the usual row index) and `j` to the flat index of that matrix' component (instead of the column index). I'd rather use `i_matrix` and `i_component`. This also applies to `Assign<vector<Matrix>>`.

Reply:
done :)